services/worker/src/worker/job_runners/dataset/loading_tags.py

```diff
@@ -41,19 +575,35 @@ def compute_loading_tags_response(dataset: str) -> DatasetLoadingTagsResponse:
     dataset_info_best_response = get_previous_step_or_raise(kinds=["dataset-info"], dataset=dataset)
     http_status = dataset_info_best_response.response["http_status"]
     tags: list[DatasetLoadingTag] = []
+    python_loading_methods: list[PythonLoadingMethod] = []
+    infos: list[dict[str, Any]] = []
+    builder_name: Optional[str] = None
     if http_status == HTTPStatus.OK:
         try:
             content = dataset_info_best_response.response["content"]
-            infos = list(islice(content["dataset_info"].values(), CROISSANT_MAX_CONFIGS))
-            if infos:
-                tags.append("croissant")
-                if infos[0]["builder_name"] == "webdataset":
-                    tags.append("webdataset")
+            infos = list(islice(content["dataset_info"].values(), LOADING_METHODS_MAX_CONFIGS))
         except KeyError as e:
             raise PreviousStepFormatError(
                 "Previous step 'dataset-info' did not return the expected content.", e
             ) from e
-    return DatasetLoadingTagsResponse(tags=tags)
+    if infos:
+        # mlcroissant library
+        python_loading_methods.append(get_python_loading_method_for_mlcroissant_library(dataset, infos))
+        tags.append("croissant")
+        # datasets library
+        python_loading_methods.append(get_python_loading_method_for_datasets_library(dataset, infos))
+        tags.append("hf_datasets")
+        # pandas or dask or webdataset library
+        builder_name = infos[0]["builder_name"]
+        if builder_name in get_python_loading_method_for_builder:
+            try:
+                python_loading_method = get_python_loading_method_for_builder[builder_name](dataset, hf_token)
```

Comment:
Currently we return only one builder-specific loading method: pandas OR dask OR webdataset

This loading method is added to the list of loading methods that can also include `datasets` and `mlcroissant`

Reply:
OK. So: why not return an array so that if we change our mind or want to return two values for a specific case, we won't have to change the code in moonlanding?

Reply:
The jobs does return an array ;)

```python
python_loading_methods: list[PythonLoadingMethod]
```

Reply:
wow ok, sorry about that :)